src/nwbinspector/nwbinspector.py

```diff
@@ -523,31 +528,36 @@ def inspect_nwb(
     filterwarnings(action="ignore", message="No cached namespaces found in .*")
     filterwarnings(action="ignore", message="Ignoring cached namespace .*")
 
-    with pynwb.NWBHDF5IO(path=nwbfile_path, mode="r", load_namespaces=True, driver=driver) as io:
-        if not skip_validate:
-            validation_errors = pynwb.validate(io=io)
-            for validation_error in validation_errors:
+    if nwbfile_object is not None:
+        for inspector_message in run_checks(nwbfile=nwbfile_object, checks=checks):
+            inspector_message.file_path = nwbfile_path
+            yield inspector_message
+    else:
```

Comment:
Can you try adjusting your usage of the Inspector to essentially just do these 3-4 lines instead of increasing the [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) of a particular function?